app/models/ability.rb

```diff
@@ -66,4 +63,17 @@ def initialize(user)
       allowed
     end
   end
+
+  private
+
+  def admin_owner_or_anon_run(user, run)
+    if user.admin?
+      true
+    elsif user.new_record?
+      # We're currently anonymous
+      run.user_id.nil?
+    else
+      run.user_id == user.id
```

Comment:
It is worth noting that this is a bit unusual:
Anonymous users can access anonymous runs, but logged in users cannot access anonymous runs.
This is probably a good approach so a logged in user doesn't start working on some anonymous run, but it should be documented here and perhaps in a test if not already.

Reply:
this is now documented with tests